overlord/devicestate/devicestate.go

```diff
@@ -1415,6 +1415,11 @@ func Remodel(st *state.State, new *asserts.Model, localSnaps []*snap.SideInfo, p
 		}
 	}
 
+	// check for exclusive changes again since we released the lock
+	if err := snapstate.CheckChangeConflictRunExclusively(st, "remodel"); err != nil {
```

Comment:
giving up here is a bit expensive otoh this is probably rare enough that it doesn't matter, OTOH if we had a TODO to always do remodelTasks from inside an embryonic change (as we always do for the most complex remodels) this check would not be needed. Anyway just some considerations

Reply:
Could you elaborate on what you mean by "embryonic change"? I'm not familiar with that terminology.